rfcs/geometric-primitives.md

```diff
@@ -0,0 +1,419 @@
+| Plane     | ✔ | ❌ | ✔ Plane |
+| Torus     | ✔ | ❌ | ❌ |
+
+### Meshing
+
+Both 2d and 3d primitives can implement the `Meshable` trait to provide the ability to generate a mesh from the primitive's definition.
+
+```rust
+let circle_mesh: Mesh = Circle2d::new(2.0).mesh();
+```
+The base primitive types only define the shape and size of the geometry about the origin. From there, the mesh generated from the primitive can have a transform applied to it like any other mesh.
+
+## Bounding
+
+For use in physics, spatial queries, collisions, raycast, or other similar use cases, `BoundVol` and `BoundArea` traits are provided. These traits are implemented for primitive types that define position and orientation in addition to the shape and size defined in the base types. For example, a 
+Geometric primitives have a defined shape, size, position, and orientation. Position and orientation are **not** defined using bevy's `Transform` components. This is because these are fundamental geometric primitives that must be usable and comparable as-is.
+
+`bevy_geom` provides two main modules, `geom_2d` and `geom_3d`. Recall that the purpose of this crate is to provide lightweight types, so there are what appear to be duplicates in 2d and 3d, such as `geom_2d::Line2d` and `geom_3d::Line`. Note that the 2d version of a line is lighter weight, and is only defined in 2d. 3d geometry (or 2d with depth which is 3d) is assumed to be the default for most cases. The names of the types were chosen with this in mind, to guide you toward using Line instead of Line2d for example, unless you know why you are making this choice.
+
+## Implementation strategy
+
+### Helper Types
+
+```rust
+/// Stores an angle in radians, and supplies builder functions to prevent errors (from_radians, from_degrees)
+struct Angle(f32);
+```
+### Traits
+
+```rust
+/// A trait for types that can be used to generate a [Mesh].
+trait Meshable{
+  fn mesh(&self, subdivisions: u8) -> Mesh;
+};
+
+/// Convex geometric primitives that are fully defined in 3D space, and can be used to bound entities within their volume. [BoundingVolume]s are useful for raycasting, collision detection, physics, and building BVHs.
+trait BoundingVolume {
+  fn raycast(&self, ray: Ray) -> Option(Point);
+}
+
+/// Convex geometric primitives that are fully defined in 2D space, and can be used to bound entities within their area. Similar to [BoundingVolumes] but for 2D instead of 3D space.
+trait BoundingArea {
+  fn raycast(&self, ray: Ray2d) -> Option(Point2d);
+}
+```
+
+### 3D Geometry Types
+
+```rust
+/// Point in 3D space
+struct Point(Vec3)
+
+/// Vector direction in 3D space that is guaranteed to be normalized through its getter/setter.
+struct Direction(Vec3)
+
+// Plane in 3d space defined by a point on the plane, and the normal direction of the plane
+struct Plane {
+  point: Point,
+  normal: Direction,
+}
+impl Meshable for Plane {}
+
+/// Unbounded line in 3D space with directionality
+struct Axis { 
```

Comment:
This could be renamed to Line to keep it consistent with Line2d (thus _Line_ would always mean unbounded)
```suggestion
struct Line { 
```